frontend/packages/kubevirt-plugin/src/selectors/vm-like/next-run-changes.ts

```diff
@@ -0,0 +1,161 @@
+
+    if (diskEquality) {
+      const volWrapper = new VolumeWrapper(vol);
+      return volWrapper.areVolumesEqual(vmiVolLookup[vol.name], true);
+    }
+
+    return false;
+  });
+};
+
+export const isCDROMChanged = (vm: VMWrapper, vmi: VMIWrapper): boolean => {
+  if (vm.isEmpty() || vmi.isEmpty()) {
+    return false;
+  }
+  return isDisksChanged(vm, vmi, vm.getCDROMs(), vmi.getCDROMs());
+};
+
+export const isBootOrderChanged = (vm: VMWrapper, vmi: VMIWrapper): boolean => {
+  if (vm.isEmpty() || vmi.isEmpty()) {
+    return false;
+  }
+
+  const vmBootOrder: BootableDeviceType[] = getBootableDevicesInOrder(vm.asResource(true));
+  const vmiBootOrder: BootableDeviceType[] = getVMIBootableDevicesInOrder(vmi.asResource(true));
+
+  if (vmBootOrder.length !== vmiBootOrder.length) {
+    return true;
+  }
+
+  // implicit boot order
+  if (vmBootOrder.length === 0) {
+    const vmDevices = getDevices(vm.asResource());
+    const vmiDevices = getVMIDevices(vmi.asResource());
+
+    let isDevsEqual = true;
+    vmDevices.forEach((bootableDevice, index) => {
+      if (!_.isEqual(bootableDevice, vmiDevices[index])) {
+        isDevsEqual = false;
+      }
+    });
+
+    if (!isDevsEqual) {
+      return true;
+    }
+  }
+
+  return !vmBootOrder.every(
+    (device, index) =>
+      device.type === vmiBootOrder[index].type &&
+      device.typeLabel === vmiBootOrder[index].typeLabel &&
+      device.value.bootOrder === vmiBootOrder[index].value.bootOrder &&
+      device.value.name === vmiBootOrder[index].value.name,
+  );
+};
+
+export const isNicsChanged = (vm: VMWrapper, vmi: VMIWrapper): boolean => {
+  if (vm.isEmpty() || vmi.isEmpty()) {
+    return false;
+  }
+
+  const vmNics = vm.getNetworkInterfaces();
+  const vmiNics = vmi.getNetworkInterfaces();
+
+  if (vmNics.length !== vmiNics.length) {
+    return true;
+  }
+
+  const vmNicsLookup = createBasicLookup(vmNics, getSimpleName);
+  const vmiNicsLookup = createBasicLookup(vmiNics, getSimpleName);
+
+  return !Object.keys(vmNicsLookup).every(
+    (vmNicName) =>
+      !!vmiNicsLookup[vmNicName] && _.isEqual(vmiNicsLookup[vmNicName], vmNicsLookup[vmNicName]),
+  );
+};
+
+export const isEnvDisksChanged = (vm: VMWrapper, vmi: VMIWrapper): boolean => {
+  if (vm.isEmpty() || vmi.isEmpty()) {
+    return false;
+  }
+
+  const vmEnvDiskVolumeNames = vm
+    .getVolumes()
+    .filter((vol) => new VolumeWrapper(vol).getType().isEnvType())
+    .map((vol) => vol.name);
+
+  const vmiEnvDiskVolumeNames = vmi
+    .getVolumes()
+    .filter((vol) => new VolumeWrapper(vol).getType().isEnvType())
+    .map((vol) => vol.name);
+
+  if (vmEnvDiskVolumeNames.length !== vmiEnvDiskVolumeNames.length) {
+    return true;
+  }
+  const vmEnvDisks = vm.getDisks().filter((dsk) => vmEnvDiskVolumeNames.includes(dsk.name));
+  const vmiEnvDisks = vmi.getDisks().filter((dsk) => vmiEnvDiskVolumeNames.includes(dsk.name));
+
+  return isDisksChanged(vm, vmi, vmEnvDisks, vmiEnvDisks);
```

Comment:
can we use passing of just the names like we talked about?

Reply:
that's not going to be enough cause for the envDisks the names aren't enough - I have to get the disks anyway.

Reply:
I know, but it would be a simpler interface and you can use disks or volumes indiscriminately